student-store-ui/src/components/ProductGrid/ProductGrid.jsx

```diff
@@ -0,0 +1,97 @@
+import * as React from "react"
+import "./ProductGrid.css"
+import ProductCard from "../ProductCard/ProductCard";
+import { useState } from "react";
+export default function ProductGrid({products ,handleAddItemsToCart, handleRemoveItemToCart}) {
+  // const [loadedProd, setLoadedProd] = useState(null);
+  // const currentLoaded = products.
+  const [showMenu, setShowMenu] = useState(true);
+  const [searched, setSearched] = useState("");
+  const regexp = new RegExp(searched, "i")
+  const [clickedCategory, setClickedCategory] = useState("");
+  let currSearch =  products.filter((products) => regexp.test(products.name))
+  let currProds = products.filter((product, idx) => product.category == clickedCategory);
+  const handleShowMenu = () =>{
+    setShowMenu(!showMenu)
+  }
+  const handleClick = (category) => {
+    setClickedCategory(category);
+  }
+  const handleSearched = (event) => {
+    const searched = event.target.value
+    setSearched(searched);
+  };
+  return (
+    <section className="products">
+      
+      <div id="cover">
+      <form method="get" onChange={(event) => handleSearched(event)}>
+        <div className="tb">
+          <div className="td">
+              <input className="search-text" type="text" required/>
+          </div>
+          <div className="td" id="s-cover">
+            <button className="search" type="submit">
+              <div id="s-circle"></div>
+              <span></span>
+            </button>
+          </div>
+        </div>
+      </form>
+    </div>
+    {showMenu == true
+      ?<div className="menu">
+      <ul className="category-menu open">
+      <button class="toggle-menu" onClick={() => handleShowMenu()}>
+        <i className="material-icons">menu</i>
+      </button>
+          <li>
+              <button className="menu-btn" onClick={() => handleClick("")}>All Categories</button>
+          </li>
+          <li>
+              <button className="menu-btn" onClick={() => handleClick("clothing")}>Clothing</button>
+          </li>
+          <li>
+              <button className="menu-btn" onClick={() => handleClick("food")}>Food</button>
+          </li>
+          <li>
+              <button className="menu-btn" onClick={() => handleClick("accessories")}>Accessories</button>
+          </li>
+          <li>
+              <button className="menu-btn" onClick={() => handleClick("tech")}>Tech</button>
+          </li>
```

Comment:
nit - can call the setter directly setClickedCategory()

also make reusable component to render these with ["", "clothing",... ].map()